File: spring/src/main/java/org/sw/sample/exception/NotFoundException.java
package org.sw.sample.exception;

import org.springframework.http.HttpStatus;

public class NotFoundException extends CommonException {
    public static final String NOT_FOUND_PATH = "Not Found Path";
    private final String reason;

    private NotFoundException(String reason, String message) {
        super(HttpStatus.NOT_FOUND, message);
        this.reason = reason;
    }

    @Override
    public String getReason() {
        return this.reason;
    }

    /**
     * 요청한 데이터를 찾을 수 없을 때 사용
     *
     * @param message 에러 메시지
     * @return exception
     */
    public static NotFoundException ofPath(String message) {
        return new NotFoundException(NOT_FOUND_PATH, "요청하신 경로를 찾을 수 없습니다: " + message);
    }


}

File: kotlin/src/main/kotlin/org/sw/spring/kotlin/exception/NotFoundException.kt
package org.sw.spring.kotlin.exception

import jakarta.websocket.CloseReason
import org.springframework.http.HttpStatus

open class NotFoundException private constructor(
    title: String,
    detail: String,
    instance: String,
    cause: Throwable? = null
) : CommonException(
    HttpStatus.NOT_FOUND, title, detail, instance, cause
) {

    companion object {
        const val NOT_FOUND_PATH = "Not Found Path"

        fun ofPath(detail: String, cause: Throwable? = null): NotFoundException {
            return NotFoundException(NOT_FOUND_PATH, detail, "/NotFoundPage", cause)
        }
    }
}